4.1.0

zzak opened the pull request:
2a813b3 is the meat of this release

Commit:
Ignore patch-level version when selecting data to install

Commit:
Remove vim syntax and require 'rubygems'

Commit:
Remove rubyforge option from spec and add me as developer

Commit:
Remove `:install_rdoc` task, and use the system rdoc instead

Commit:
Implement supported ruby version build step to replace multiruby

This allows us to drop dependency on third-party tools to build the data.

In order to build the data, you need the necessary tools for building Ruby.

This change also modifies the build system to use a different naming scheme.

Previously, multiruby was used to select the version build dir, then rdoc was
run on each version of ruby installed from multiruby to generate the ri data.

The new system works like this:

For each SUPPORTED_VERSIONS:
  * Download the tarball (i.e. ruby-2.3.0.tar.gz)
  * Extract and move it without patch-level (i.e. rubies/2.3)
  * Generate the data in matching format (i.e. data/2.3)

This allows us to then select the data more easily using only RUBY_VERSION.

We need to strip any patch-level from the version to find the data during install.

See this commit for more: fd4c6554e4fe95ac78631495a73c4100cd72c6af

Commit:
Update rdoc link

Commit:
Ignore local rubies build dir

Commit:
Bump to version 4.1.0

hsbt commented on Rakefile:
```diff
@@ -1,12 +1,10 @@
-require 'rubygems'
 require 'hoe'
 
+SUPPORTED_VERSIONS = %w[1.8.7-p374 1.9.3-p551 2.0.0-p647 2.1.8 2.2.3 2.3.0]
```
2.2.4?

Commit:
Add executable bit for bin/rdoc-data

Commit:
Add license to gemspec to kill warning

Commit:
Use latest 2.2 patch-level

zzak commented:
Here's the current blocker.. and it's got something to do with 1.8.

```
~ => ruby -v
ruby 1.8.7 (2013-12-22 patchlevel 375) [x86_64-linux]

~ => rdoc-data --install

~ => ri Object
/home/zzak/.gem/ruby/1.8.7/gems/rdoc-4.2.1/lib/rdoc/store.rb:548:in `load': undefined class/module Encoding (ArgumentError)
        from /home/zzak/.gem/ruby/1.8.7/gems/rdoc-4.2.1/lib/rdoc/store.rb:548:in `load_cache'
        from /home/zzak/.gem/ruby/1.8.7/gems/rdoc-4.2.1/lib/rdoc/store.rb:547:in `open'
        from /home/zzak/.gem/ruby/1.8.7/gems/rdoc-4.2.1/lib/rdoc/store.rb:547:in `load_cache'
        from /home/zzak/.gem/ruby/1.8.7/gems/rdoc-4.2.1/lib/rdoc/ri/driver.rb:391:in `initialize'
        from /home/zzak/.gem/ruby/1.8.7/gems/rdoc-4.2.1/lib/rdoc/ri/paths.rb:54:in `each'
        from /home/zzak/.gem/ruby/1.8.7/gems/rdoc-4.2.1/lib/rdoc/ri/driver.rb:385:in `initialize'
        from /home/zzak/.gem/ruby/1.8.7/gems/rdoc-4.2.1/lib/rdoc/ri/driver.rb:362:in `new'
        from /home/zzak/.gem/ruby/1.8.7/gems/rdoc-4.2.1/lib/rdoc/ri/driver.rb:362:in `run'
        from /home/zzak/.gem/ruby/1.8.7/gems/rdoc-4.2.1/bin/ri:12
        from /home/zzak/.gem/ruby/1.8.7/bin/ri:23:in `load'
        from /home/zzak/.gem/ruby/1.8.7/bin/ri:23
```

**but** if I clear the ri cache.. it works:

```
ruby -rubygems -e 'require "rdoc"; require "rdoc/ri/paths"; system "rm #{RDoc::RI::Paths.system_dir}/cache.ri"'
```

drbrain commented:
Did you build the data with Ruby 1.8.7? Can you dump the cache.ri with ri --dump

zzak commented:
@drbrain I think just using the system rdoc, but can change that to use the same version of Ruby that we're building.

Also, `ri --dump` doesn't work for me:

```
ri --dump
/home/zzak/.rubies/1.8.7-p375/lib/ruby/1.8/optparse.rb:451:in `parse': missing argument: --dump (OptionParser::MissingArgument)
```

drbrain commented:
You need to give it the file to dump (path to cache.ri)

Commit:
:scissors:

Commit:
Refactor `:build_ri_data` into each version task

Now each version also creates a `destdir` used for the install prefix.

From here, we can use each version of ruby, and rdoc, to generate the ri data.

This change also adds a `:clean` task to remove any build artifacts.

Commit:
Use File.exists? incase building with older rubies

Commit:
Refactor to use directory task for `:rubies` and `:data`,
as well as `rdoc_dir`, or output dir, and `buildir` (where ruby is built).

Commit:
Use generate script to setup options for newer RDoc.

This avoids a bug when using the command-line rdoc form,
but still works on older versions of RDoc.

zzak commented:
@drbrain Almost ready to wrap this up, just doing some testing to make sure all versions are working properly. Then I should be able to ship the gem!

zzak commented:
Now.. for some reason, the 1.8 dir isn't being included when building the gem.

```
~/code/rdoc-data => ls data/
1.8  1.9  2.0  2.1  2.2  2.3
~/code/rdoc-data => ls pkg/rdoc-data-4.1.0/data/
1.8  1.9  2.0  2.1  2.2  2.3
```

from the gem:

```
~/.gem/ruby/1.8.7/gems => ls rdoc-data-4.1.0/data/
1.9  2.0  2.1  2.2  2.3
```

Commit:
:nail_care:

Commit:
Ensure gem_path is valid

Commit:
Hacks on hacks to get a modern rdoc for Ruby 1.8.7
